### PR TITLE
nRF54l20 soc and board

### PR DESCRIPTION
**Pull request by kl-cruz**
Add device and board descriptions.

### CODE REVIEW

File: boards/nordic/nrf54l20pdk/Kconfig [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# nRF54L15 PDK board configuration
```
Comment (hubertmis) [inline boards/nordic/nrf54l20pdk/Kconfig:1]: typo: 54L20

File: dts/common/nordic/nrf54l20.dtsi [hunk 2]
```diff
@@ -0,0 +1,574 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <mem.h>
+#include <nordic/nrf_common.dtsi>
+
+/delete-node/ &sw_pwm;
+
+/* Domain IDs. Can be used to specify channel links in IPCT nodes. */
+#define NRF_DOMAIN_ID_APPLICATION 0
```
Comment (hubertmis) [inline dts/common/nordic/nrf54l20.dtsi:13]: There is no IPCT in nRF54L20. Is there?

**Review by hubertmis — APPROVED** [review]

**Review by e-rk — APPROVED** [review]

**Review by anangl — CHANGES_REQUESTED** [review]
- commit "[nrf fromlist] manifest: Update hal_nordic with nrfx 3.6" is invalid; there is no such commit in https://github.com/zephyrproject-rtos/zephyr/pull/76332. Instead, commits from https://github.com/zephyrproject-rtos/zephyr/pull/76330 should be cherry-picked as fromtree ones
- commit "[nrf noup] drivers: entropy: Switch entropy driver in nRF54L20" should have the following line as the first one in its message body:
`nrf-squash! 
[nrf noup] entropy: Add fake entropy nRF PRNG driver`

**Comment by kl-cruz** [comment]
Closing as https://github.com/nrfconnect/sdk-zephyr/pull/2000 includes newer changes

**Pull request closed without merging**